src/nodes/chunk_append/planner.c

```diff
@@ -369,37 +365,27 @@ ts_chunk_append_get_scan_plan(Plan *plan)
 				return (Scan *) plan;
 			}
 
-			if (strcmp(custom->methods->CustomName, VECTOR_AGG_NODE_NAME) == 0)
-			{
-				/*
-				 * This is a vectorized aggregation node, we have to recurse
-				 * into its child, similar to the normal aggregation node.
-				 *
-				 * Unfortunately we have to hardcode the node name here, because
-				 * we can't depend on the TSL library.
-				 */
+			/* CustomScan nodes keep child plan nodes in custom_plans. For
+			 * nodes with one child, recurse into the child to find a baserel
+			 * scan. It is not clear what a CustomScan with multiple children
+			 * would mean here so we don't handle it. */
+			if (list_length(custom->custom_plans) == 1)
```

Comment:
This part is not needed for the fix? I'd leave it as is, not sure that recursing into arbitrary custom scan nodes is always correct for the ChunkAppend optimization. Technically this might even be a plan node from another extension.

Reply:
Yes, it is not needed to avoid the error. But why would it not be correct? It is a recursive call that will just do nothing if it doesn't find a baserel plan node it recognizes for the chunk append purpose.